contracts/STYLE_GUIDE.md

```diff
@@ -0,0 +1,347 @@
+- Immutable variables should be prefixed with an `i_` to make it clear that they are immutable. E.g. `i_decimalPlaces`. They should always be private, and you should write explicit getters if you want to expose an immutable variable.
+- Internal/private constants should be all caps with underscores: `FOO_BAR`. Like other contract variables, constants should not be public. Create getter methods if you want to publicly expose constants.
+- Explicitly declare variable size: `uint256` not just `uint`. In addition to being explicit, it matches the naming used to calculate function selectors.
+- Mapping should always be named if Solidity allows it (≥0.8.18)
+
+### Types
+
+- If you are storing an address and know/expect it to be of a type(or interface), make the variable that type. This more clearly documents the behavior of this variable than the `address` type and often leads to less casting code whenever the address is used.
+
+### Structs
+
+- All structs should be packed to have the lowest memory footprint to reduce gas usage. Even structs that will never be written to storage should be packed.
+  - A contract can be considered a struct; it should also be packed to reduce gas cost.
+- Structs should contain struct packing comments to clearly indicate the storage slot layout
+  - Using the exact characters from the example below will ensure visually appealing struct packing comments.
+  - Notice there is no line on the unpacked last `fee` item.
+- Struct should contain comments, clearly indicating the denomination of values e.g. 0.01 USD if the variable name doesn’t already do that (which it should).
+  - Simple tool that could help packing structs and adding comments: https://github.com/RensR/Spack
+
+```solidity
+/// @dev Struct to hold the fee configuration for a fee token, same as the FeeTokenConfig but with
+/// token included so that an array of these can be passed in to setFeeTokenConfig to set the mapping
+struct FeeTokenConfigArgs {
+  address token; // ────────────╮ Token address
+  uint32 networkFeeUSD; //      │ Flat network fee to charge for messages, multiples of 0.01 USD
+  //                            │ multiline comments should work like this. More fee info 
+  uint64 gasMultiplier; // ─────╯ Price multiplier for gas costs, 1e18 based so 11e17 = 10% extra cost
+  uint64 premiumMultiplier; // ─╮ Multiplier for fee-token-specific premiums
+  bool enabled; // ─────────────╯ Whether this fee token is enabled
+  uint256 fee; //                 The flat fee the user pays in juels        
+}
+```
+
+# Functions
+
+### Visibility
+
+- Method visibility should always be explicitly declared. Contract’s [public interfaces should be tested](https://github.com/smartcontractkit/chainlink/blob/master/contracts/test/test-helpers/helpers.ts#L221) to enforce this and make sure that internal logic is not accidentally exposed.
```

Comment:
`All contract variables should be private by default`

It will depend on the exact contract, if you the variable in higher layers, it has to be internal. Defaulting to private is the safest option